packages/astro/src/content/utils.ts

```diff
@@ -117,6 +118,37 @@ export async function getEntryData(
 				message: AstroErrorData.ContentSchemaContainsSlugError.message(entry.collection),
 			});
 		}
+
+		/**
+		 * Resolve all the images referred to in the frontmatter from the file requesting them
+		 */
+		async function preprocessAssetPaths(object: Record<string, any>) {
+			if (typeof object !== 'object' || object === null) return;
+
+			for (let [schemaName, schema] of Object.entries<any>(object)) {
+				if (schema._def.description === '__image') {
+					object[schemaName] = z.preprocess(
+						async (value: unknown) => {
+							if (!value || typeof value !== 'string') return value;
+							return (await resolver(value))?.id;
```

Comment:
Do we have a test for aliased paths?

Reply:
We do! [This file](https://github.com/withastro/astro/blob/8b2af558b481015a93100b1811f4c85959a42ce0/packages/astro/test/fixtures/core-image-ssg/src/content/blog/one.md) has it all: a relative path, a nested path and an aliased path.